README.md

```diff
@@ -1,17 +1,167 @@
 # Name
+
 meta-lua-nginx-module - templates and toolchains for generating
 http-lua-nginx-module and stream-lua-nginx-module.
 
-This module is experimental and should not be used in production yet.
-
 # Usage
+
 ```shell
-$ make SUBSYSTEM=http
+$ make SUBSYS=stream DESTDIR=/path/to/stream-lua-nginx-module/src -j4
+```
+
+All make options are optional.
+
+## Supported Options
+
+| Option        | Meaning                                             | Default     |
+| ------------- | --------------------------------------------------- | ----------- |
+| SUBSYS        | Subsystem this build is for                         | `http`      |
+| DESTDIR       | Directory the generated files should go             | `build/src` |
+
+# Directory Structures
 ```
+├── src: all source files
+│   ├── http: http subsystem specific files, not rendered through the template engine
+│   ├── stream: stream subsystem specific files, not rendered through the template engine
+│   └── subsystem: templates file that are shared between subsystems, rendered the template engine
+└── utils: build toolchains
+```
+# Developing
+## General Workflow
+
+First, determine where the change should goes to. As an example, if you would like to change
+a file inside `lua-nginx-module`, that file could either be rendered through the template engine,
+or be simply copied from the `src/http` directory as-is without going through the template engine.
+
+To tell which one is the case, look at the first few lines of the generated file. A file rendered by
+the template engine will have something like this:
+
+```C
+/*
+ * !!! DO NOT EDIT, YOUR CHANGES WILL BE OVERWRITTEN !!!
+ * Generated from: src/subsystem/ddebug.h.tt2
+ */
```

Comment:
Just a side note: I think it's better to generate this comment header in mini-tt2 directly.